src/derivatives/05-diffusion-derivatives.md

```diff
@@ -41,83 +41,81 @@ which has its own unique requirements with respect to:
     empirical image data;
 
 -   The appropriate [data representations](#data-representations) utilized to
-    store information parameterised [by](#paramdef-fit) or
-    [from](#paramdef-derived) the model onto the filesystem;
+    store information about model results in the filesystem;
 
 -   The requirements for encapsulation and complete representation of
     derived [orientation information](#orientation-specification), which is
     a key strength of diffusion MRI but presents unique challenges for
     correct interpretation.
 
-### Parameter terminology
 
-Throughout this document, the term "parameter" is used to refer to
-multiple distinct sources of information. The distinction between
-these uses is defined thus:
-
-1.  <a name="paramdef-input">*Input*</a> parameter:
-
-    Value or non-numerical setting that influences the conformation
-    of the diffusion model to the empirical diffusion-weighted data.
-
-1.  <a name="paramdef-fit">Model *fit*</a> parameter:
+### File names
 
-    Value that is the direct result of fitting the diffusion model to the
-    empirical diffusion-weighted data.
+When a model is specified by the data, parameters of the model can be saved
+using the following conventions:
 
-1.  <a name="paramdef-derived">Model-*derived*</a> parameter:
+```Text
+<pipeline_name>/
+    sub-<participant_label>/
+        dwi/
+            <source_keywords>[_space-<space>]_model-<label1>_param-<param1>_dwimap.nii[.gz]
+            <source_keywords>[_space-<space>]_model-<label1>_param-<param2>_dwimap.nii[.gz]
+            <source_keywords>[_space-<space>]_model-<label1>_dwimap.json
+            <source_keywords>[_space-<space>]_model-<label2>_param-<parama>_dwimap.nii[.gz]
+            <source_keywords>[_space-<space>]_model-<label2>_param-<paramb>_dwimap.nii[.gz]
+            <source_keywords>[_space-<space>]_model-<label2>_dwimap.json
+```
 
-    Value that can be calculated directly from previously estimated
-    *model parameters*, without necessitating reference to
-    the empirical diffusion-weighted data.
+- Files "`<source_keywords>[_space-<space>]_model-<label>_param-<param*>_dwimap.nii[.gz]`"
+    provide model outputs and/or parameters of the model.
 
-For example, consider a diffusion tensor model fit: the number of
-iterations in the optimization algorithm would be an *input* parameter;
-the six unique diffusion tensor coefficients would be the *model fit*
-parameters; the Fractional Anisotropy (FA) would be a *model-derived*
-parameter (as it is calculated from the diffusion tensor coefficients
-rather than the image data).
+- Only one metadata file, "`<source_keywords>[_space-<space>]_model-<label>_dwimap.json`"
+    is allowed for each set of files that has a specific `model-<label>` entity.
```

Comment:
This isn't workable as the rest of the BEP currently stands. Take for example the ball-and-sticks model (just the mean fit, ignoring the bootstrap realisations). You have some images that yield a scalar per voxel, and then you have the stick orientations, which are natively spherical coordinates. The specification needs to provide a way with which to encode the fact that the appropriate interpretation of the former is as a scalar parameter, whereas the latter is as spherical orientations, with a specific reference frame.

(While it's technically true that here you could distinguish between these two cases based on the dimensionality of the image, eg. 3D image = Scalar, 4D image with 2 volumes = spherical orientations, it doesn't extend to more complex cases. Mixture models could have all sorts of different components with different ways of encoding anisotropic information. Indeed you could just have different scalar model parameters that have different units. Then there's the fact that if you don't distinguish between model fit and model-derived parameters, you need to be able to avoid such ambiguities / conflicts for all possible derivative parameters.)

Previously I've tried to do this through complex inheritance. There would be one JSON file encoding everything that applied to all model parameters, and then a separate JSON for any parameter that required additional information necessary for the interpretation of those data that don't apply to other parameters from the same model. If you want to force a single metadata file per model, you would need to within that file define a dictionary containing parameter names and their corresponding metadata. While it is not my preference, since it would represent quite a fundamental change to how metadata is specified in BIDS (as opposed to complex inheritance which is just extension of an existing mechanism), it would be at least functional.

Reply:
Yes - the intent is that the one json for the model would have sub-dicts, which can be used to refer to different `dwimap`s. I'll clarify.